### PR TITLE
Make watchdog print every 5 seconds when it's frozen

### PR DESCRIPTION
**Pull request by hafarooki**
too tired to rewrite this

### CODE REVIEW

**Comment by hafarooki** [comment]
https://gist.github.com/Miclebrick/b157aaee5e6020581dd5de43cc42e07f proof it works

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
++                } else {
++                    log.log( Level.SEVERE, "The server has not responded for " + shortTimeout / 1000 + " seconds! Creating thread dump");
++                }
++                // Paper end - Different message for short timeout
+                 log.log( Level.SEVERE, "------------------------------" );
+                 log.log( Level.SEVERE, "Server thread dump (Look for plugins here before reporting to Paper!):" );
+                 dumpThread( ManagementFactory.getThreadMXBean().getThreadInfo( MinecraftServer.getServer().primaryThread.getId(), Integer.MAX_VALUE ), log );
+                 log.log( Level.SEVERE, "------------------------------" );
+                 //
++                if(isLongTimeout) { // Paper - Only print full dump on long timeouts
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch]: code style

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
++                if( isLongTimeout )
++                {
+                 log.log( Level.SEVERE, "The server has stopped responding!" );
+                 log.log( Level.SEVERE, "Please report this to https://github.com/PaperMC/Paper/issues" );
+                 log.log( Level.SEVERE, "Be sure to include ALL relevant console errors and Minecraft crash reports" );
+@@ -79,11 +92,16 @@ public class WatchdogThread extends Thread
+                     }
+                 }
+                 // Paper end
++                } else {
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch]: code style

**Comment by hafarooki** [comment]
is that really necessary

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+     }
+ 
+     public static void doStart(int timeoutTime, boolean restart)
+@@ -52,9 +55,19 @@ public class WatchdogThread extends Thread
+         while ( !stopping )
+         {
+             //
+-            if ( lastTick != 0 && System.currentTimeMillis() > lastTick + timeoutTime && !Boolean.getBoolean("disable.watchdog")) // Paper - Add property to disable
++            long currentTime = System.currentTimeMillis(); // Paper - do we REALLY need to call this method multiple times?
++            if ( lastTick != 0 && currentTime > lastTick + shortTimeout && !Boolean.getBoolean("disable.watchdog")) // Paper - Add property to disable and short timeout
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch]: please add a space before the closing )

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                 // Paper end
++                } else {
++                    log.log( Level.SEVERE, "The server has not responded for " + shortTimeout / 1000 + " seconds! Creating thread dump");
++                }
++                // Paper end - Different message for short timeout
+                 log.log( Level.SEVERE, "------------------------------" );
+                 log.log( Level.SEVERE, "Server thread dump (Look for plugins here before reporting to Paper!):" );
+                 dumpThread( ManagementFactory.getThreadMXBean().getThreadInfo( MinecraftServer.getServer().primaryThread.getId(), Integer.MAX_VALUE ), log );
+                 log.log( Level.SEVERE, "------------------------------" );
+                 //
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch:63]: wats this

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+         this.timeoutTime = timeoutTime;
+         this.restart = restart;
++        shortTimeout = Math.min(5000, timeoutTime); // Paper - Make short timeout the lower of 5 seconds, and timeout time
+     }
+ 
+     public static void doStart(int timeoutTime, boolean restart)
+@@ -52,9 +55,19 @@ public class WatchdogThread extends Thread
+         while ( !stopping )
+         {
+             //
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch:33]: wats this

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch (hunk 3 above)
Comment (hafarooki) [reply Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch]: it wasn't even there before

Comment (Miclebrick) [reply Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch]: it wasn't even there before

Comment (Spottedleaf) [reply Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch]: fits with style

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch (hunk 4 above)
Comment (hafarooki) [reply Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch:63]: dunno, i didn't put it there

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch (hunk 5 above)
Comment (hafarooki) [reply Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch:33]: dunno, i didn't put it there

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
++                if( isLongTimeout )
++                {
+                 log.log( Level.SEVERE, "The server has stopped responding!" );
+                 log.log( Level.SEVERE, "Please report this to https://github.com/PaperMC/Paper/issues" );
+                 log.log( Level.SEVERE, "Be sure to include ALL relevant console errors and Minecraft crash reports" );
+@@ -79,11 +92,18 @@ public class WatchdogThread extends Thread
+                     }
+                 }
+                 // Paper end
++                } else {
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch:55]: code style

Comment (hafarooki) [reply Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch:55]: you're kidding

**Comment by willies952002** [comment]
Maybe merge the 3 commits into one? :thinking:

**Comment by zachbr** [comment]
We can handle that ourselves at merge time

File: Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch (hunk 6 above)
Comment (Spottedleaf) [reply Spigot-Server-Patches/0349-Add-5-second-short-dumps-to-watchdog.patch:55]: Use this

`}`
`else`
`{`

**Comment by zachbr** [comment]
b369e70970322d15e928f8b4d1135578e0f9c58a

**Comment by mibby** [comment]
@zachbr Is there any way to disable this so it isn't enabled by default? I get this error every time I start my server.

```
[05:53:11] [Paper Watchdog Thread/ERROR]: The server has not responded for 5 seconds! Creating thread dump
[05:53:11] [Paper Watchdog Thread/ERROR]: ------------------------------
[05:53:11] [Paper Watchdog Thread/ERROR]: Server thread dump (Look for plugins here before reporting to Paper!):
[05:53:11] [Paper Watchdog Thread/ERROR]: ------------------------------
[05:53:11] [Paper Watchdog Thread/ERROR]: Current Thread: Server thread
[05:53:11] [Paper Watchdog Thread/ERROR]: 	PID: 19 | Suspended: false | Native: true | State: RUNNABLE
[05:53:11] [Paper Watchdog Thread/ERROR]: 	Stack:
[05:53:11] [Paper Watchdog Thread/ERROR]: 		java.net.SocketInputStream.socketRead0(Native Method)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		java.net.SocketInputStream.read(SocketInputStream.java:171)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		java.net.SocketInputStream.read(SocketInputStream.java:141)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.InputRecord.read(InputRecord.java:503)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:983)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:940)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		java.io.BufferedInputStream.read(BufferedInputStream.java:345)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:735)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:678)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1587)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.mojang.authlib.HttpAuthenticationService.performPostRequest(HttpAuthenticationService.java:87)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService.makeRequest(YggdrasilAuthenticationService.java:66)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.mojang.authlib.yggdrasil.YggdrasilGameProfileRepository.findProfilesByNames(YggdrasilGameProfileRepository.java:54)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.destroystokyo.paper.profile.PaperGameProfileRepository.findProfilesByNames(PaperGameProfileRepository.java:45)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull$1.load(TileEntitySkull.java:64)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull$1.load(TileEntitySkull.java:46)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3716)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2424)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2298)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2211)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache.get(LocalCache.java:4154)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4158)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5147)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5153)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull$3.call(TileEntitySkull.java:206)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull$3.call(TileEntitySkull.java:203)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull.b(TileEntitySkull.java:222)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.inventory.CraftMetaSkull.applyToItem(CraftMetaSkull.java:86)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.inventory.CraftItemStack.setItemMeta(CraftItemStack.java:389)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.inventory.CraftItemStack.asNMSCopy(CraftItemStack.java:46)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.inventory.CraftInventory.setItem(CraftInventory.java:96)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.utils.SkinLoader.lambda$loadPlayerHeadSkins$1(SkinLoader.java:48)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.utils.SkinLoader$$Lambda$611/833862246.accept(Unknown Source)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		java.util.ArrayList.forEach(ArrayList.java:1257)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.utils.SkinLoader.loadPlayerHeadSkins(SkinLoader.java:48)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.Main.lambda$null$0(Main.java:49)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.Main$$Lambda$601/86037990.onRetrieve(Unknown Source)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.mysql.DbSkinLoader.lambda$null$0(DbSkinLoader.java:41)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.mysql.DbSkinLoader$$Lambda$608/619026031.run(Unknown Source)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:63)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:423)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:839)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:423)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:773)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665)
[05:53:11] [Paper Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:748)
[05:53:11] [Paper Watchdog Thread/ERROR]: ------------------------------

[05:53:16] [Paper Watchdog Thread/ERROR]: The server has not responded for 5 seconds! Creating thread dump
[05:53:16] [Paper Watchdog Thread/ERROR]: ------------------------------
[05:53:16] [Paper Watchdog Thread/ERROR]: Server thread dump (Look for plugins here before reporting to Paper!):
[05:53:16] [Paper Watchdog Thread/ERROR]: ------------------------------
[05:53:16] [Paper Watchdog Thread/ERROR]: Current Thread: Server thread
[05:53:16] [Paper Watchdog Thread/ERROR]: 	PID: 19 | Suspended: false | Native: true | State: RUNNABLE
[05:53:16] [Paper Watchdog Thread/ERROR]: 	Stack:
[05:53:16] [Paper Watchdog Thread/ERROR]: 		java.net.SocketInputStream.socketRead0(Native Method)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		java.net.SocketInputStream.read(SocketInputStream.java:171)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		java.net.SocketInputStream.read(SocketInputStream.java:141)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.InputRecord.read(InputRecord.java:503)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:983)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:940)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		java.io.BufferedInputStream.read(BufferedInputStream.java:345)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:735)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:678)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1587)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.mojang.authlib.HttpAuthenticationService.performPostRequest(HttpAuthenticationService.java:87)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService.makeRequest(YggdrasilAuthenticationService.java:66)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.mojang.authlib.yggdrasil.YggdrasilGameProfileRepository.findProfilesByNames(YggdrasilGameProfileRepository.java:54)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.destroystokyo.paper.profile.PaperGameProfileRepository.findProfilesByNames(PaperGameProfileRepository.java:45)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull$1.load(TileEntitySkull.java:64)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull$1.load(TileEntitySkull.java:46)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3716)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2424)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2298)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2211)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache.get(LocalCache.java:4154)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4158)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5147)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5153)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull$3.call(TileEntitySkull.java:206)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull$3.call(TileEntitySkull.java:203)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.TileEntitySkull.b(TileEntitySkull.java:222)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.inventory.CraftMetaSkull.applyToItem(CraftMetaSkull.java:86)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.inventory.CraftItemStack.setItemMeta(CraftItemStack.java:389)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.inventory.CraftItemStack.asNMSCopy(CraftItemStack.java:46)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.inventory.CraftInventory.setItem(CraftInventory.java:96)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.utils.SkinLoader.lambda$loadPlayerHeadSkins$1(SkinLoader.java:48)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.utils.SkinLoader$$Lambda$611/833862246.accept(Unknown Source)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		java.util.ArrayList.forEach(ArrayList.java:1257)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.utils.SkinLoader.loadPlayerHeadSkins(SkinLoader.java:48)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.Main.lambda$null$0(Main.java:49)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.Main$$Lambda$601/86037990.onRetrieve(Unknown Source)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.mysql.DbSkinLoader.lambda$null$0(DbSkinLoader.java:41)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		com.live.bemmamin.pocketgames.mysql.DbSkinLoader$$Lambda$608/619026031.run(Unknown Source)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:63)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:423)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:839)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:423)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:773)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665)
[05:53:16] [Paper Watchdog Thread/ERROR]: 		java.lang.Thread.run(Thread.java:748)
[05:53:16] [Paper Watchdog Thread/ERROR]: ------------------------------
```

**Comment by sgdc3** [comment]
I think it would be better to enable this dump only after 30sec / 1minute from startup.

**Comment by hafarooki** [comment]
welcome to PR that change yourself :joy:

**Comment by hafarooki** [comment]
@mibby your skin loader thing should be doing that async.... if at all??

**Comment by hafarooki** [comment]
Does the watchdog start after startup or during startup? If it's after, that's the fault of the plugins for loading things after onenable IMO. Might be good to add a config option for it regardless

**Comment by aikar** [comment]
Yes I think we should delay monitoring until after server is ready, as it is common to by desire, block the main to ensure critical resources are all finished loading, even if that means a 15s+ startup delay.

Please open an issue @mibby

**Pull request closed without merging**